### PR TITLE
bug(rhineng-9037): Resolve case insensitive filtering issue

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-9037](https://issues.redhat.com/browse/RHINENG-9037).
* Update fliters to perform case insensitive queries for identified fields.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
LGTM

File: api/filtering/db_filters.py [hunk 1]
```diff
@@ -249,16 +253,16 @@ def query_filters(
...
     if fqdn:
-        filters += _canonical_fact_filter("fqdn", fqdn)
+        filters += _canonical_fact_filter("fqdn", fqdn, case_insensitive=True)
     elif display_name:
         filters += _display_name_filter(display_name)
     elif hostname_or_id:
         filters += _hostname_or_id_filter(hostname_or_id)
     elif insights_id:
-        filters += _canonical_fact_filter("insights_id", insights_id)
+        filters += _canonical_fact_filter("insights_id", insights_id, case_insensitive=True)
```
Comment (vkrizan) [inline api/filtering/db_filters.py:262]: Making `insights_id` case insensitive would prevent the index to be used. Being fixed via #1905

**Pull request merged**